### ISSUE TITLE
Add $wp_object property for Timber core objects

### ISSUE THREAD

**Issue by gchtr** [timber/timber#2560]
**Ticket: [#2501](https://github.com/timber/timber/issues/2501#issuecomment-948792581)**

In various cases we need to get properties from WordPress core objects. For example, in a post, we need to use the `WP_Post` object to maybe overwrite the `$post` global for better compatibility within loops (#2545).

Or we have to access properties like `post_title` from the original post. Instead of saving all the values in separate properties in a Timber object, we could introduce a new `$wp_object` property for Timber core entities like `Timber\Post` or `Timber\MenuItem`.

In https://github.com/timber/timber/issues/2501#issuecomment-948792581, @nlemoine made a nice example for this:

```php
<?php

class Post {
    protected WP_post $wp_object = $post;

    public function core_object() : WP_Post {
        return $wp_object;
    }

    public function title() {
        return $wp_object->post_title;
    }

    public function date() {
        return new DateTimeImmutable( $wp_object->post_date );
    }

    // etc.
}
```

**Comment by acobster** [comment]
I've often wanted something like this and I like this proposal. The only suggestion I have would be to back it with an interface. Making this update in `CoreInterface` would be another breaking change, but that might be the best place for it.

**Comment by gchtr** [comment]
Resolved in #2615.

Using the `$wp_object` property in more places requires us to first move away from so many public properties (see https://github.com/timber/timber/pull/2615#issuecomment-1198998173). And that is something to handle in 2.x-future.
